client/src/stream.rs

```diff
@@ -212,8 +211,9 @@ impl rpc::Server for CallbackServer {
                 };
 
                 self.duplex_input = if let StreamDirection::Duplex = self.dir {
-                    match Vec::try_with_capacity(shm_area_size) {
-                        Ok(duplex_input) => Some(duplex_input),
+                    let mut duplex_input = Vec::new();
+                    match duplex_input.try_reserve_exact(shm_area_size) {
```

Comment:
Prefer using `try_reserve_exact` since `CallbackReq::SharedMem` is called once when receiving `ServerMessage::StreamCreate`?

Reply:
Yeah, and we know the exact allocation size up front.